subprojects/platform-native/src/main/java/org/gradle/nativeplatform/toolchain/internal/gcc/GccPlatformToolProvider.java

```diff
@@ -138,6 +152,11 @@
         return new ArStaticLibraryArchiver(buildOperationExecutor, commandLineTool(staticLibArchiverTool), context(staticLibArchiverTool), workerLeaseService);
     }
 
+    private GccVersionResult gccMetadata(GccCommandLineToolConfigurationInternal cppCompilerTool) {
+        File gccLocation = toolSearchPath.locate(cppCompilerTool.getToolType(), cppCompilerTool.getExecutable()).getTool();
```

Comment:
What if we can't find the tool?

Reply:
If we are at this point, the tool needs to be found. The toolchain is available at this point. `commandLineTool` uses the same call.

Reply:
Ah, I see, we throw a detailed exception. That should work fine.